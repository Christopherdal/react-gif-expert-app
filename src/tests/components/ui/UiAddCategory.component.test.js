import React from 'react';
import { shallow } from 'enzyme';
import { UiAddCategory } from './../../../shared/components/ui/UiAddCategory.component';

describe('Test on UiAddCategory Ui Component', () => {
	const setCategories = jest.fn();
	// let wrapper = shallow(<UiAddCategory setCategories={setCategories} />); // Just for get de autofill
	let wrapper;

	beforeEach(() => {
		jest.clearAllMocks(); // Clear simulates to every test
		wrapper = shallow(<UiAddCategory setCategories={setCategories} />);
	});

	test('should Match With Snapshot UiAddCategory Correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should change box text', () => {
		const input = wrapper.find('input');
		const value = 'Hello world';
		input.simulate('change', { target: { value } });
		expect(wrapper.find('h1').text().trim()).toBe(value);
	});

	test('should Not change box text', () => {
		wrapper.find('form').simulate('submit', { preventDefault() {} });

		expect(setCategories).not.toHaveBeenCalled();
	});
});
