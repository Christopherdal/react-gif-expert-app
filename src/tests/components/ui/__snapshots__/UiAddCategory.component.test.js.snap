// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on UiAddCategory Ui Component should Match With Snapshot UiAddCategory Correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1 />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
